icu4c/source/common/uloc.cpp

```diff
@@ -1746,9 +1739,13 @@ uloc_getScript(const char*    localeID,
     }
 
     if(_isIDSeparator(*localeID)) {
-        i=ulocimp_getScript(localeID+1, script, scriptCapacity, NULL);
+        return ulocimp_getScript(localeID+1, NULL, *err).extract(script, scriptCapacity, *err);
     }
-    return u_terminateChars(script, scriptCapacity, i, err);
+
+    if (script != NULL && scriptCapacity > 0) {
```

Comment:
The old code set a not-terminated-warning if the capacity is 0. How about keeping `return u_terminateChars(script, scriptCapacity, 0, err);` instead of this replacement if-terminate-return?

Reply:
Done.